api/server/graph.cc

```diff
@@ -608,9 +608,19 @@ std::string Graph::fw_build_rule(const app::Rule &rule) {
 std::string Graph::fw_build_sg(const app::Sg &sg) {
     std::string r;
     for (auto it = sg.rules.begin(); it != sg.rules.end();) {
-        r += "(" + fw_build_rule(it->second) + ")";
+        auto fw_rule = fw_build_rule(it->second);
+        if (fw_rule.length() == 0) {
+            it++;
+            continue;
+        }
+        r += "(" + fw_rule + ")";
         if (++it != sg.rules.end())
-            r += " || ";
+            r += "||";
+    }
+    // Special case when last rule is empty
+    if (r.back() == '|') {
```

Comment:
how could this happen ? if a rule is empty you skip the rule with a continue ...